src/test/libcephfs/flock.cc

```diff
@@ -175,27 +178,27 @@ static void thread_ConcurrentLocking(str_ConcurrentLocking& s) {
   ASSERT_GE(fd, 0); 
 
   ASSERT_EQ(-EWOULDBLOCK,
-	    ceph_flock(cmount, fd, LOCK_EX | LOCK_NB, pthread_self()));
+	    ceph_flock(cmount, fd, LOCK_EX | LOCK_NB, (uint64_t)pthread_self()));
```

Comment:
assuming `pthread_self()` returns `uint64_t` is wrong. but this is also incorrect. i'd suggest add 
```c++
uint64_t get_owner() {
  auto me = pthread_self();
  static_assert(std::is_convertible_v<decltype(me), uint64_t> ||
                std::is_pointer_v<decltype(me)>,
                "we need to use pthread_self() for the owner parameter");
  return reinterpret_cast<uint64_t>(me);
}
```
to document this assumption, and s/pthread_self/get_owner/ in the tests.

Reply:
how about putting this code snippet in this test.

Reply:
@tchaikov 
For now I have it in `compat.{h,cc}`.
Since it will allow to "fix" the other usuage.

Reply:
i don't it's **right** to do so. as it's not related to compatibility, actually it defeats the purpose of compatibility. per the POSIX standard, `pthread_t` is an opaque type, which could be a pointer, an integer or a structure. it's us who are using the fact that it's an `unsigned long` in glibc.

Reply:
@tchaikov 
That is true....
I'll throw it into `src/test/libcephfs/`